File: java-pkmst/src/main/java/com/prokarma/pkmst/model/RequestRoutesProperty.java
package com.prokarma.pkmst.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonValue;
/**
 * Response class to be returned by Api
 * @author pkmst
 *
 */
import com.fasterxml.jackson.annotation.JsonCreator;

/**
 * Gets or Sets RequestRoutesProperty
 */
public enum RequestRoutesProperty {
  
  TRAVEL_TIME("travel_time"),
  
  DISTANCE("distance"),
  
  FARES("fares"),
  
  ROUTE("route");

  private String value;

  RequestRoutesProperty(String value) {
    this.value = value;
  }

  @Override
  @JsonValue
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static RequestRoutesProperty fromValue(String text) {
    for (RequestRoutesProperty b : RequestRoutesProperty.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + text + "'");
  }
}

File: kotlin-spring/src/main/kotlin/traveltimeplatform/models/RequestRoutesProperty.kt
package traveltimeplatform.models

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonValue
import javax.validation.Valid
import javax.validation.constraints.*

/**
* 
* Values: travelTime,distance,fares,route
*/
enum class RequestRoutesProperty(val value: String) {

    travelTime("travel_time"),

    distance("distance"),

    fares("fares"),

    route("route");

}

